internal/tiltfile/value/cmd.go

```diff
@@ -0,0 +1,72 @@
+package value
+
+import (
+	"fmt"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+
+	"github.com/tilt-dev/tilt/internal/tiltfile/starkit"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+const typeCmd = "cmd"
+
+type CmdPlugin struct{}
+
+func NewCmdPlugin() CmdPlugin {
+	return CmdPlugin{}
+}
+
+var _ starkit.Plugin = CmdPlugin{}
+
+func (p CmdPlugin) OnStart(env *starkit.Environment) error {
+	if err := env.AddBuiltin("cmd", p.cmd); err != nil {
+		return fmt.Errorf("could not add cmd builtin: %v", err)
+	}
+	return nil
+}
+
+func (p CmdPlugin) cmd(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple,
+	kwargs []starlark.Tuple) (starlark.Value, error) {
+	var argsVal, argsBatVal starlark.Value
+	// TODO(milas): should we also except ['FOO=bar'] style list?
+	var env StringStringMap
+	var dir starlark.Value
+
+	err := starkit.UnpackArgs(
+		thread, fn.Name(), args, kwargs,
+		"args", &argsVal,
+		"env?", &env,
+		"dir?", &dir,
```

Comment:
IIUC, custom_build ignores its command's dir and env:
https://github.com/tilt-dev/tilt/blob/2b9a8539f211bd9441c54a7109b78d53e7c5cbaa/internal/build/custom_builder.go#L77-L79
https://github.com/tilt-dev/tilt/blob/333ce8741d1b6285add1625f3ed004e8290d14f8/internal/tiltfile/docker.go#L361

(really, the CustomBuild struct probably shouldn't have a WorkDir field - I assume that predates Command having other fields)

It'd be great to fix, but also maybe a bigger yak than you want in this PR, in which case maybe add an error if someone passes a `cmd()` to `custom_build`

Reply:
on the plus side, yay for this change making it more likely we have consistent support for cmd customization across builtins!